lib/screens/home/home_view_model.dart

```diff
@@ -69,26 +85,55 @@ class HomeViewModel extends AutoDisposeAsyncNotifier<void> {
   }
 
   Future<void> getSurveyList() async {
+    _pageNumber += 1;
+    if (_numberOfPage != null && _pageNumber > (_numberOfPage ?? 0)) {
+      return;
+    }
+    _getSurveysFromNetwork();
+
     if (_surveyCache.isEmpty) {
       showOrHideShimmer(true);
     } else {
       showOrHideLoadMore(true);
     }
+  }
```

Comment:
@kaungkhantsoe I mean
```suggestion
    _pageNumber += 1;
    if (_numberOfPage != null && _pageNumber > (_numberOfPage ?? 0)) {
      return;
    }

    if (_surveyCache.isEmpty) {
      showOrHideShimmer(true);
    } else {
      showOrHideLoadMore(true);
    }
     _getSurveysFromNetwork();
     showOrHideShimmer(false);
    showOrHideLoadMore(false);
  }
```

Reply:
@manh-t if we move like that, loadings will not show since they will be executed in order without waiting.

https://github.com/nimblehq/flutter-ic-kaung-thieu/assets/32578035/01d76f62-4c77-4623-904b-dfcfd8a289f8

Reply:
@kaungkhantsoe it is because you didn't put await to the get survey function: `await _getSurveysFromNetwork();`

Reply:
@manh-t  Fixed in https://github.com/nimblehq/flutter-ic-kaung-thieu/pull/79/commits/c1b9c1b907500576808c33cb2d1fbff3ef9574c1